webrtc/README.md

```diff
@@ -0,0 +1,316 @@
+
+  Yes, with WebRTC's datachannels running on top of SCTP, there is no need for
+  additional multiplexing.
+
+## General Open Questions
+
+- Should libp2p's WebRTC stack limit itself to using UDP only, or support WebRTC
+  on top of both UDP and TCP?
+
+- Is the fact that Firefox does not allow a WebRTC to `localhost` an issue? See
+  https://bugzilla.mozilla.org/show_bug.cgi?id=831926.
+
+## Previous, ongoing and related work
+
+- Proof of concept for the server side in rust-libp2p:
+  https://github.com/libp2p/rust-libp2p/pull/2622
+
+- Proof of concept for the server side (native) and the client side (Rust in
+  WASM): https://github.com/wngr/libp2p-webrtc
+
+- WebRTC using STUN and TURN: https://github.com/libp2p/js-libp2p-webrtc-star
+
+# FAQ
+
+- _Why exchange the TLS certificate fingerprint in the multiaddr? Why not
+  base it on the libp2p public key?_
+
+  Browsers do not allow loading a custom certificate. One can only generate a
+  certificate via
+  [rtcpeerconnection-generatecertificate](https://www.w3.org/TR/webrtc/#dom-rtcpeerconnection-generatecertificate).
+
+- _Why not embed the peer ID in the TLS certificate, thus rendering the
+  additional "peer certificate" exchange obsolete?_
+
+  Browsers do not allow editing the properties of the TLS certificate.
+
+- _How about distributing the multiaddr in a signed peer record, thus rendering
+  the additional "peer certificate" exchange obsolete?_
+
+  Signed peer records are not yet rolled out across the many libp2p protocols.
+  Making the libp2p WebRTC protocol dependent on the former is not deemed worth
+  it at this point in time. Later versions of the libp2p WebRTC protocol might
+  adopt this optimization.
+
+  Note, one can role out a new version of the libp2p WebRTC protocol through a
+  new multiaddr protocol, e.g. `/webrtc-2`.
+
+- _Why exchange fingerprints in an additional authentication handshake on top of
+  an established WebRTC connection? Why not only exchange signatures of ones TLS
+  fingerprints signed with ones libp2p private key on the plain WebRTC
+  connection?_
+
+  Once _A_ and _B_ established a WebRTC connection, _A_ sends
+  `signature_libp2p_a(fingerprint_a)` to _B_ and vice versa. While this has the
+  benefit of only requring two messages, thus one round trip, it is prone to a
+  key compromise and replay attack. Say that _E_ is able to attain
+  `signature_libp2p_a(fingerprint_a)` and somehow compromise _A_'s TLS private
+  key, _E_ can now impersonate _A_ without knowing _A_'s libp2p private key.
+
+  If one requires the signatures to contain both fingerprints, e.g.
+  `signature_libp2p_a(fingerprint_a, fingerprint_b)`, the above attack still
+  works, just that _E_ can only impersonate _A_ when talking to _B_.
+
+  Adding a cryptographic identifier of the unique connection (i.e. session) to
+  the fingerprint would protect against this attack. To the best of our
+  knowledge the browser does not give us access to such identifier.
```

Comment:
As far as I know the browser does not give us access to a unique cryptographic session identifier after the DTLS handshake.  @melekes @ckousik @marten-seemann  please correct me if I am wrong.


 If that were the  case, I *think* (don't trust my expertise) we could do without the additional Noise handshake and instead do a simple one round trip signature exchange.